### PR TITLE
[4.16-4.18] use release channel to replace prerelease image config setting

### PR DESCRIPTION
**Pull request by jianlinliu**
There are several places using `prerelease.version_bounds` to set `release:latest` or `release:target` for getting a stable build, to be consistent, replace `prerelease.version_bounds` with `release.channel` image config setting. That also helps easy-reading.

### CODE REVIEW

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-multi-stable-4.18-upgrade-from-stable-4.17-aws-ipi-proxy-sts-arm-f28

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-stable-4.18-upgrade-from-stable-4.17-vsphere-ipi-proxy-fips-f28

**Comment by jianlinliu** [comment]
From job log, release:latest and target:latest can get the latest stable build image successfully.

[periodic-ci-openshift-openshift-tests-private-release-4.18-multi-stable-4.18-upgrade-from-stable-4.17-aws-ipi-proxy-sts-arm-f28](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/56899/rehearse-56899-periodic-ci-openshift-openshift-tests-private-release-4.18-multi-stable-4.18-upgrade-from-stable-4.17-aws-ipi-proxy-sts-arm-f28/1836597056215453696)
```
INFO[2024-09-19T02:44:02Z] Requesting 4.17 from https://api.integration.openshift.com/api/upgrades_info/graph?arch=multi&channel=candidate-4.17 
INFO[2024-09-19T02:44:02Z] Resolved release latest to quay.io/openshift-release-dev/ocp-release@sha256:18fed9398469b38d4f443361a3dca6db46279a7807cef95c2f5170d1a10edcef 

# oc adm release info quay.io/openshift-release-dev/ocp-release@sha256:18fed9398469b38d4f443361a3dca6db46279a7807cef95c2f5170d1a10edcef
Name:           4.17.0-rc.3
Digest:         sha256:18fed9398469b38d4f443361a3dca6db46279a7807cef95c2f5170d1a10edcef
Created:        2024-09-16T13:36:00Z
OS/Arch:        multi (linux/amd64)
Manifests:      734
```
[periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-stable-4.18-upgrade-from-stable-4.17-vsphere-ipi-proxy-fips-f28](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/56899/rehearse-56899-periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-stable-4.18-upgrade-from-stable-4.17-vsphere-ipi-proxy-fips-f28/1836597338156568576)

```
INFO[2024-09-19T02:45:06Z] Requesting 4.18 from https://api.integration.openshift.com/api/upgrades_info/graph?arch=amd64&channel=candidate-4.18 
INFO[2024-09-19T02:45:06Z] Resolved release target to quay.io/openshift-release-dev/ocp-release@sha256:e930176c384d1c819f6259c403cb9a4fe419eebb1da80a9ae0a0e8afbd74d964 

# oc adm release info quay.io/openshift-release-dev/ocp-release@sha256:e930176c384d1c819f6259c403cb9a4fe419eebb1da80a9ae0a0e8afbd74d964
Name:           4.18.0-ec.0
Digest:         sha256:e930176c384d1c819f6259c403cb9a4fe419eebb1da80a9ae0a0e8afbd74d964
Created:        2024-09-02T08:13:09Z
OS/Arch:        linux/amd64
Manifests:      742
Metadata files: 1

```

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-stable-4.16-upgrade-from-stable-4.15-azure-ipi-usertags-sa-encryption-arm-f28

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-multi-nightly-4.17-upgrade-from-stable-4.17-gcp-ipi-xpn-oidc-arm-f60 periodic-ci-openshift-openshift-tests-private-release-4.17-multi-stable-4.17-upgrade-from-stable-4.16-gcp-ipi-compact-filestore-csi-arm-f28 periodic-ci-openshift-openshift-tests-private-release-4.17-arm64-nightly-4.17-upgrade-from-stable-4.17-baremetal-upi-ovn-ipv4-f60

**Comment by jianlinliu** [comment]
4.6:

[periodic-ci-openshift-openshift-tests-private-release-4.16-multi-stable-4.16-upgrade-from-stable-4.15-azure-ipi-usertags-sa-encryption-arm-f28](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/56899/rehearse-56899-periodic-ci-openshift-openshift-tests-private-release-4.16-multi-stable-4.16-upgrade-from-stable-4.15-azure-ipi-usertags-sa-encryption-arm-f28/1836609026469662720)

```
INFO[2024-09-19T03:31:36Z] Requesting 4.15 from https://api.integration.openshift.com/api/upgrades_info/graph?arch=multi&channel=candidate-4.15 
INFO[2024-09-19T03:31:36Z] Resolved release latest to quay.io/openshift-release-dev/ocp-release@sha256:412cb889a50812978da822fb33801f6ce974df546a527739e3ec8c7dc5594a8b 

# oc adm release info quay.io/openshift-release-dev/ocp-release@sha256:412cb889a50812978da822fb33801f6ce974df546a527739e3ec8c7dc5594a8b
Name:           4.15.33
Digest:         sha256:412cb889a50812978da822fb33801f6ce974df546a527739e3ec8c7dc5594a8b
Created:        2024-09-16T17:17:53Z
OS/Arch:        multi (linux/amd64)
Manifests:      698
Metadata files: 1
```

**Comment by jianlinliu** [comment]
4.17:

[periodic-ci-openshift-openshift-tests-private-release-4.17-arm64-nightly-4.17-upgrade-from-stable-4.17-baremetal-upi-ovn-ipv4-f60](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/56899/rehearse-56899-periodic-ci-openshift-openshift-tests-private-release-4.17-arm64-nightly-4.17-upgrade-from-stable-4.17-baremetal-upi-ovn-ipv4-f60/1836610095274790912)
```
INFO[2024-09-19T03:35:50Z] Requesting 4.17 from https://api.integration.openshift.com/api/upgrades_info/graph?arch=arm64&channel=candidate-4.17 
INFO[2024-09-19T03:35:51Z] Resolved release arm64-latest to quay.io/openshift-release-dev/ocp-release@sha256:d41d55c8c22dc840684be536b7353b44457899db7a941b11eb98218c002dab0f 
 
INFO[2024-09-19T03:35:51Z] Requesting 4.17 from https://api.integration.openshift.com/api/upgrades_info/graph?arch=amd64&channel=candidate-4.17 
INFO[2024-09-19T03:35:51Z] Resolved release latest to quay.io/openshift-release-dev/ocp-release@sha256:8073209e3aff4fef0f2f705a3868da3c56e85575010f713802f4ef651127180a 

# oc adm release info quay.io/openshift-release-dev/ocp-release@sha256:d41d55c8c22dc840684be536b7353b44457899db7a941b11eb98218c002dab0f
Name:           4.17.0-rc.3
Digest:         sha256:d41d55c8c22dc840684be536b7353b44457899db7a941b11eb98218c002dab0f
Created:        2024-09-16T13:35:59Z
OS/Arch:        linux/arm64
Manifests:      732
Metadata files: 1


# oc adm release info quay.io/openshift-release-dev/ocp-release@sha256:8073209e3aff4fef0f2f705a3868da3c56e85575010f713802f4ef651127180a
Name:           4.17.0-rc.3
Digest:         sha256:8073209e3aff4fef0f2f705a3868da3c56e85575010f713802f4ef651127180a
Created:        2024-09-16T13:35:59Z
OS/Arch:        linux/amd64
Manifests:      734
Metadata files: 1
```

[periodic-ci-openshift-openshift-tests-private-release-4.17-multi-stable-4.17-upgrade-from-stable-4.16-gcp-ipi-compact-filestore-csi-arm-f28](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/56899/rehearse-56899-periodic-ci-openshift-openshift-tests-private-release-4.17-multi-stable-4.17-upgrade-from-stable-4.16-gcp-ipi-compact-filestore-csi-arm-f28/1836610095169933312)
```
INFO[2024-09-19T03:35:51Z] Requesting 4.16 from https://api.integration.openshift.com/api/upgrades_info/graph?arch=multi&channel=candidate-4.16 
INFO[2024-09-19T03:35:51Z] Resolved release latest to quay.io/openshift-release-dev/ocp-release@sha256:e08c4fd2d215783acda7f5956fcacc0093584d9afcd0e3070113b1263a4bcf21


# oc adm release info quay.io/openshift-release-dev/ocp-release@sha256:e08c4fd2d215783acda7f5956fcacc0093584d9afcd0e3070113b1263a4bcf21
Name:           4.16.13
Digest:         sha256:e08c4fd2d215783acda7f5956fcacc0093584d9afcd0e3070113b1263a4bcf21
Created:        2024-09-16T17:11:53Z
OS/Arch:        multi (linux/amd64)
Manifests:      728
Metadata files: 1
```

**Comment by jianlinliu** [comment]
cc @jiajliu @dis016 to review.

/pj-rehearse ack

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__multi-stable-4.16-upgrade-from-stable-4.15.yaml [hunk 1]
```diff
@@ -42,12 +42,10 @@ build_root:
...
 releases:
   latest:
-    prerelease:
+    release:
       architecture: multi
-      product: ocp
-      version_bounds:
-        lower: 4.15.0-0
-        upper: 4.16.0-0
+      channel: candidate
```
Comment (jiajliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__multi-stable-4.16-upgrade-from-stable-4.15.yaml:47]: how about using `fast` channel as start version in upgrade jobs? `fast` channel is the supported channel officially while `candidate` is not.

Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__multi-stable-4.16-upgrade-from-stable-4.15.yaml]: Done, also update other ci yaml files together.

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-stable-4.16-upgrade-from-stable-4.15-azure-ipi-usertags-sa-encryption-arm-f28

**Comment by jianlinliu** [comment]
cc @liangxia to review

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__arm64-stable-4.18-upgrade-from-stable-4.17.yaml [hunk 2]
```diff
@@ -47,26 +47,20 @@ releases:
...
-    prerelease:
+    release:
       architecture: amd64
-      product: ocp
-      version_bounds:
-        lower: 4.18.0-0
-        stream: 4-dev-preview
-        upper: 4.19.0-0
+      channel: candidate
+      version: "4.18"
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__arm64-stable-4.18-upgrade-from-stable-4.17.yaml:63]: Do we have the EC build here for 4.18 already ?

Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__arm64-stable-4.18-upgrade-from-stable-4.17.yaml:63]: Yeah, refer to https://github.com/openshift/release/pull/56899#issuecomment-2359882809

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__arm64-stable-4.18-upgrade-from-stable-4.17.yaml:63]: Ack. Good to know.

**Pull request merged**